src/walk_dirs.rs

```diff
@@ -403,4 +406,33 @@ mod test {
             discovered
         );
     }
+
+    #[test]
+    fn test_retrieve_emtpy_imports_on_ignored_dirs_match() {
+        let discovered: Vec<WalkFileData> = discover_fences_and_files(
+            "tests/walk_dir_simple",
+            crate::walk_dirs::ExternalFences::Ignore,
+            vec![regex::Regex::new("tests/.**/subdir").unwrap()],
+        );
+
+        let expected_subdir_ts_file = SourceFile {
+            source_file_path: "tests/walk_dir_simple/subdir/subsubdir/subSubDirFile.ts".to_owned(),
+            tags: set!(
+                "root-fence-tag-1".to_owned(),
+                "root-fence-tag-2".to_owned(),
+                "subsubdir-fence-tag".to_owned()
+            ),
+            imports: HashMap::new(),
+        };
+
+        assert!(
+            discovered.iter().any(|x| match x {
+                WalkFileData::SourceFile(y) => expected_subdir_ts_file == *y,
+                _ => false,
+            }),
+            "expected discovered files to contain {:?}, but it did not. Actual: {:?}",
+            expected_subdir_ts_file,
+            discovered
+        );
+    }
```

Comment:
add test to validate that fence.json files in ignored directories are also ignored.